build-bot-status.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+import sys
+import time
+import threading
+
+import requests
+
+from light_controller import LightController
+
+
+class BuildBotSession:
+    failColour = "red"
+    busyColour = "orange"
+    goodColour = "green"
+    exceptionColour = "purple"
+
+    def __init__(self, builderURL, slaveName):
+        self.builderURL = builderURL
+        self.slaveName = slaveName
+        self._statusURL = "{}/json/builders/{}".format(self.builderURL,
+                                                       self.slaveName)
+
+        self.lightController = LightController("/dev/ttyACM0",
+                                               "data/buttons.json")
```

Comment:
please compose LightController with class than strongly coupling with BuildBotSession. Testing would become easy.

Reply:
https://medium.com/swlh/the-best-way-to-understand-composition-in-python-5-case-studies-and-solution-4b23a6a2cc38